content/blog/from-cms-to-contextual.md

```diff
@@ -0,0 +1,424 @@
+      body
+    }
+  }
+}`
+```
+
+### Adding `useTina` hook
+
+The `useTina` hook is used to make a piece of Tina content editable. It is code-split so in production, this hook will pass through the data value. When you are in edit mode, it registers an editable form in the sidebar. 
+
+### Adding imports
+
+The first thing that needs to be done is import useTina, useEffect and useState. We can also remove a few imports that we will no longer be using. 
+
+```bash
+import {useState, useEffect} from 'react'
+import {useTina} from 'tinacms/dist/edit-state'
+```
+
+### Changing our props and adding UseTina
+
+We are going to update our props from `({ post, morePosts, preview })` to `(props)` . We are going to pass the props into our `useTina` hook. Now that has been updated with the props coming in we can use `useTina` . 
+
+```jsx
+const { data } = useTina({
+    query,
+    variables: props.variables,
+    data: props.data,
+  }) 
+```
+
+As you can see, we are reusing our query from before and passing the variables, and data to `useTina`.  This now means we can power our site using contextual editing. Congratulations your site now has superpowers! 
+
+### Update our elements to use Tina data
+
+Now we have access to our Tina powered data we can go through and update all of our elements to use Tina. You can replace each of the `post.` with `data.getPostsDocument.data.` and then replace the `!post?.slug` with `!props.variables.relativePath` when you are finished your return code should look like:
+
+```jsx
+const router = useRouter()
+  if (!router.isFallback && !props.variables.relativePath) {
+    return <ErrorPage statusCode={404} />
+  }
+  return (
+    <Layout preview={false}>
+      <Container>
+        <Header />
+        {router.isFallback ? (
+          <PostTitle>Loading…</PostTitle>
+        ) : (
+          <>
+            <article className="mb-32">
+              <Head>
+                <title>
+                  {data.getPostsDocument.data.title} | Next.js Blog Example with {CMS_NAME}
+                </title>
+                <meta property="og:image" content={data.getPostsDocument.data.ogImage.url} />
+              </Head>
+              <PostHeader
+                title={data.getPostsDocument.data.title}
+                coverImage={data.getPostsDocument.data.coverImage}
+                date={data.getPostsDocument.data.date}
+                author={data.getPostsDocument.data.author}
+              />
+              <PostBody content={data.getPostsDocument.data.body} />
+            </article>
+          </>
+        )}
+      </Container>
+    </Layout>
+    
+  )
+```
+
+### The final piece of Contextual Editing.
+
+One piece of code that the original was doing was taking the markdown and turning it into HTML. We currently aren’t doing that with our body, the reason is, we have to keep track of changes when using Tina. This is where useState and useEffect come in, first create a content variable that is track by a state:
```

Comment:
I don't understand this section. We can't transform markdown because we have to track changes?